frontend/src/scenes/cohorts/Cohort.tsx

```diff
@@ -1,199 +1,300 @@
-            )}
-            {cohort.id && cohort.id !== 'new' && <CohortDetailsRow cohort={cohort} />}
-            <Divider />
-            {cohort.is_static ? (
-                <div>
-                    <h3 className="l3">Add Users</h3>
-                    <span>
-                        Drop a <pre style={{ display: 'inline' }}>.csv</pre> file here to add users to your cohort
-                    </span>
-                    <Dragger {...staticCSVDraggerProps} className="cohort-csv-dragger">
-                        <p className="ant-upload-drag-icon">
-                            <InboxOutlined />
-                        </p>
-                        <div>
-                            <p className="ant-upload-text">Click or drag file to this area to upload</p>
-                            <p className="ant-upload-hint">
-                                The CSV file only requires a single column with the user’s distinct ID.
-                            </p>
+                {hasAvailableFeature(AvailableFeature.DASHBOARD_COLLABORATION) && (
+                    <Row gutter={[16, 24]} className="mt" style={{ maxWidth: 640 }}>
+                        <Col span={24}>
+                            <Field name="description" label="Description">
+                                {({ value, onValueChange }) => <LemonTextArea value={value} onChange={onValueChange} />}
+                            </Field>
+                        </Col>
+                    </Row>
+                )}
+                {cohort.is_static ? (
+                    <Row gutter={24} className="mt">
+                        <Col span={24}>
+                            <Field name="csv" label={isNewCohort ? 'Upload users' : 'Add users'}>
+                                {({ onValueChange }) => (
+                                    <>
+                                        <span className="mb">
+                                            Upload a CSV file to add users to your cohort. The CSV file only requires a
+                                            single column with the user’s distinct ID.
+                                        </span>
+                                        <Dragger
+                                            name="file"
+                                            multiple={false}
+                                            fileList={cohort.csv ? [cohort.csv] : []}
+                                            accept=".csv"
+                                            showUploadList={false}
+                                            beforeUpload={(file: UploadFile) => {
+                                                onValueChange(file)
+                                                return false
+                                            }}
+                                            className="cohort-csv-dragger"
+                                        >
+                                            {cohort.csv ? (
```

Comment:
It'd be nice if we had an 'uploading state' here. Some CSVs are pretty large, so it can take a bit.

Reply:
Dug into this a bit, and realized that we don't actually need an uploading state here!

It looks like under the hood, Antd stores the file data immediately in a File object which is a type of Blob. On reading the file's contents (onSave), the promise is executed on read, which pushing the File object into `FormData()` does implicitly. Since the actual "waiting" is offloaded to the "saving" state, we will never have an "uploading" state which is pretty neat. 

The `uploading` props in Antd's `<Dragger/>` are pretty misleading. It's not uploading to the browser per say, but actually is for the use case when you're uploading the file to a remote server (like S3 or similar).

Reply:
So I think the saving state can still take a while, but we don't handle that anywhere. You can see it if you try to upload a large CSV (depending on internet, it might help to throttle your connection).

I think the fix is something pretty simple like `{cohortLoading && <Loading />}`